robottelo/install/base.py

```diff
@@ -0,0 +1,152 @@
+# -*- encoding: utf-8 -*-
+"""Generic base class for install commands."""
+import logging
+
+from robottelo import ssh
+from robottelo.config import conf
+
+
+class InstallError(Exception):
+    """Indicates that a install command could not be run."""
+
+
+class Base(object):
+    """
+    @param command_base: base install command.
+    """
+    command_base = None  # each inherited instance should define this
+    command_requires_auth = False  # True when command requires auth user
+    command_service = None
+
+    logger = logging.getLogger('robottelo')
+
+    @classmethod
+    def reset(cls, options=None):
+        """
+        dummy
+        """
+
+        if options is None:
+            options = {}
+
+        result = cls.execute(cls._construct_command(options))
```

Comment:
You can move call to construct command to be run by execute, this way you don't need to use this structure every single time you need execute a command. This is some bad design we have in CLI and should not be followed in new code.